[gpui] Unable to run cargo run smoothly during local development, stuck at the gpui crates

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

run
```sh
% cargo run
```

Encountered a problem

```sh
error: failed to run custom build command for `gpui v0.1.0 (/Users/gowxx/code/github-upstream/zed/crates/gpui)`

Caused by:
  process didn't exit successfully: `/Users/gowxx/code/github-upstream/zed/target/debug/build/gpui-c7ef89843d0963f1/build-script-build` (exit status: 101)
  --- stdout
  cargo:rustc-link-lib=framework=System
  cargo:rerun-if-changed=src/platform/mac/dispatch.h
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-apple-darwin
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_apple_darwin
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS

  --- stderr
  src/platform/mac/dispatch.h:1:10: fatal error: 'dispatch/dispatch.h' file not found
  thread 'main' panicked at crates/gpui/build.rs:66:14:
  unable to generate bindings: ClangDiagnostic("src/platform/mac/dispatch.h:1:10: fatal error: 'dispatch/dispatch.h' file not found\n")
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
gowxx@sunbowendeMacBook-Pro zed % rustup -v
verbose: read metadata version: '12'
```

### Environment

Zed: v0.135.2 (Zed)
OS: macOS 14.5.0
Memory: 16 GiB
Architecture: aarch64

rustup 1.27.1 (54dd3d00f 2024-04-24)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
rustc 1.78.0 (9b00956e5 2024-04-29)
cargo 1.78.0 (54d8815d0 2024-03-26)

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="1273" alt="image" src="https://github.com/zed-industries/zed/assets/12870715/816be710-7026-4ca4-ae03-07a6b044fac7">

> `dispatch/dispatch.h` seems to be the file coming with XCode, have you checked the https://github.com/zed-industries/zed/blob/main/docs/src/development/macos.md#dependencies guide and installed the developer tools?

Yes, I followed the instructions in the document during the initial preparation work.

```sh
zed % xcode-select --install
xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates
zed % sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
Password:
zed % rustup target add wasm32-wasi
info: component 'rust-std' for target 'wasm32-wasi' is up to date
zed % cargo clean
     Removed 5030 files, 1.3GiB total
zed % cargo run

error: failed to run custom build command for `gpui v0.1.0 (/Users/gowxx/code/github-upstream/zed/crates/gpui)`

Caused by:
  process didn't exit successfully: `/Users/gowxx/code/github-upstream/zed/target/debug/build/gpui-c7ef89843d0963f1/build-script-build` (exit status: 101)
  --- stdout
  cargo:rustc-link-lib=framework=System
  cargo:rerun-if-changed=src/platform/mac/dispatch.h
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-apple-darwin
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_apple_darwin
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS

  --- stderr
  src/platform/mac/dispatch.h:1:10: fatal error: 'dispatch/dispatch.h' file not found
  thread 'main' panicked at crates/gpui/build.rs:66:14:
  unable to generate bindings: ClangDiagnostic("src/platform/mac/dispatch.h:1:10: fatal error: 'dispatch/dispatch.h' file not found\n")
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

--- a/docs/src/development/macos.md
+++ b/docs/src/development/macos.md
@@ -83,3 +83,31 @@ Try `sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer`
 ### Cargo errors claiming that a dependency is using unstable features
 
 Try `cargo clean` and `cargo build`.
+
+### Error: 'dispatch/dispatch.h' file not found
+
+If you encounter an error similar to:
+
+```bash
+src/platform/mac/dispatch.h:1:10: fatal error: 'dispatch/dispatch.h' file not found
+```
+
+This file is part of Xcode. Ensure you have installed the Xcode command line tools and set the correct path:
+
+```bash
+xcode-select --install
+sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
+```
+
+Additionally, set the `BINDGEN_EXTRA_CLANG_ARGS` environment variable:
+
+```bash
+export BINDGEN_EXTRA_CLANG_ARGS="--sysroot=$(xcrun --show-sdk-path)"
+```
+
+Then clean and rebuild the project:
+
+```bash
+cargo clean
+cargo run
+```

--- a/docs/src/development/macos.md
+++ b/docs/src/development/macos.md
@@ -90,6 +90,17 @@ If you encounter an error similar to:
 
 ```bash
 src/platform/mac/dispatch.h:1:10: fatal error: 'dispatch/dispatch.h' file not found
+
+Caused by:
+  process didn't exit successfully
+
+  --- stdout
+  cargo:rustc-link-lib=framework=System
+  cargo:rerun-if-changed=src/platform/mac/dispatch.h
+  cargo:rerun-if-env-changed=TARGET
+  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-apple-darwin
+  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_apple_darwin
+  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
 ```
 
 This file is part of Xcode. Ensure you have installed the Xcode command line tools and set the correct path:

